app/src/test/java/org/pesmypetcare/mypetcare/features/pets/TestPetUpdateProfileImage.java

```diff
@@ -0,0 +1,72 @@
+package org.pesmypetcare.mypetcare.features.pets;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.pesmypetcare.mypetcare.controllers.TrUpdatePetImage;
+import org.pesmypetcare.mypetcare.features.users.NotPetOwnerException;
+import org.pesmypetcare.mypetcare.features.users.User;
+import org.pesmypetcare.mypetcare.services.StubPetManagerService;
+
+import static org.junit.Assert.assertEquals;
+
+public class TestPetUpdateProfileImage {
+    private User user;
+    private Pet pet;
+    private TrUpdatePetImage trUpdatePetImage;
+
+    @Before
+    public void setUp() throws PetRepeatException {
+        user = new User("johnDoe", "johndoe@gmail.com", "1234");
+        pet = getDinkyPet();
+
+        pet.setProfileImage(getBitmap(Color.BLUE));
+        user.addPet(pet);
+
+        trUpdatePetImage = new TrUpdatePetImage(new StubPetManagerService());
+    }
+
+    @Test(expected = NotPetOwnerException.class)
+    public void shouldNotUpdateImageOfNonOwnerPet() throws NotPetOwnerException {
+        trUpdatePetImage.setUser(new User("arthurGalvin", "johngalvin@gmail.com", "5678"));
+        trUpdatePetImage.setPet(pet);
+        trUpdatePetImage.setNewPetImage(getBitmap(Color.GREEN));
+        trUpdatePetImage.execute();
+    }
+
+    @Test
+    public void shouldAddPetProfileImage() throws NotPetOwnerException {
+        Bitmap newPetImage = getBitmap(Color.GREEN);
+        trUpdatePetImage.setUser(user);
+        trUpdatePetImage.setPet(pet);
+        trUpdatePetImage.setNewPetImage(newPetImage);
+        trUpdatePetImage.execute();
+
+        assertEquals("Should add pet profileImage", newPetImage, pet.getProfileImage());
+    }
+
+    private Bitmap getBitmap(int color) {
+        Bitmap bitmap = Bitmap.createBitmap(20, 20, Bitmap.Config.ARGB_8888);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['20' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5223641)